test/test_multiprocessing.py

```diff
@@ -219,10 +220,19 @@ def _get_next_fds(self, n=1):
     def has_shm_files(self, wait=True):
         if not HAS_SHM_FILES:
             return False
+
         result = self._has_shm_files()
-        if result and mp.get_sharing_strategy() == 'file_system' and wait:
-            time.sleep(0.5)
-            return self._has_shm_files()
+        if not result or mp.get_sharing_strategy() != 'file_system' or not wait:
+            return result
+
+        total_waiting_time = 0
+        waiting_time = 0.5
+
+        while total_waiting_time <= MAX_WAITING_TIME_IN_SECONDS and result:
+            time.sleep(waiting_time)
+            total_waiting_time += waiting_time
+            result = self._has_shm_files()
+
```

Comment:
might be worth adding a log statement if self._has_shm_files() is still returning true (to help with future debugging)